### DISCUSSION TITLE
Getting g29 force feedback in a python environment

### DISCUSSION THREAD

**Discussion by limty226** [kuriatsu/ros-g29-force-feedback#31] [Q&A]
hello.

I am studying and doing research on shared control.

As shown in the video, I would like to implement autopilot on the g29 using the steering values of a virtual vehicle that is driving autonomously, but I would like to inquire whether this is possible in a Python environment.

**Comment by kuriatsu** [comment]
Yes, of course. ROS2 environment supports python3.
You will implement rospy node that receives control messages from a vehicle and sends messages to this node.
https://github.com/kuriatsu/ros-g29-force-feedback/blob/master/examples/carla_control.py
This example may be helpful for you.

**Reply by limty226** [reply]
Thank you very much for your reply.

Would that code be useful in other simulations using Python?

I would like to get force feedback from the metadrive simulation I use.

Taking control through steering operation in a vehicle in an autonomous driving environment

I am studying shared control.

**Reply by kuriatsu** [reply]
The example code receives vehicle control and rotates the steering wheel through ros message.
You can rotate the wheel following vehicle control by just changing the receiving part of the program using API from MetaDrive instead of CARLA. 
However, the function to control the vehicle with rotating the steering wheel is not implemented and I have never tested it.
I made the wheel rotate as accurately as possible, but seems not precise yet for such a closed-loop control.
It may make the car wobble.
But it is worth trying and tweaking parameters may be able to solve it!
